snapcraft/internal/parts.py

```diff
@@ -188,10 +188,11 @@ def _process_parts(self):
             if 'filesets' in properties:
                 del properties['filesets']
 
-            # FIXME: snap is deprecated, rewrite it to prime instead.
-            if properties.get('snap'):
+            # Handle the deprecated snap keyword.
+            snap = properties.pop('snap')
```

Comment:
This could be done earlier, but that would mean another sweep on the entire loaded yaml. I think it's fine to keep it here.